assets/js/stripe.js

```diff
@@ -494,10 +495,17 @@ jQuery( function( $ ) {
 			}
 
 			if ( 'card' === source_type ) {
-				stripe.handleCardPayment( client_secret, stripe_card, {
-					source_data: extra_details,
-				} )
-				.then( wc_stripe_form.intentResponse );
+				wc_stripe_form.getIntent()
+					.catch( function( error ) {
```

Comment:
The `.catch()` should be after the `.then()` steps. As it is right now, if there's an error the `.catch()` will run and after the `.then()` will be called with no parameters so `intent.client_secret` will crash.